packages/rich-text-editor/src/RichTextEditor/controlmap.ts

```diff
@@ -0,0 +1,194 @@
+import boldIcon from "@kaizen/component-library/icons/bold.icon.svg"
+import italicIcon from "@kaizen/component-library/icons/italics.icon.svg"
+import underlineIcon from "@kaizen/component-library/icons/underline.icon.svg"
+import bulletListIcon from "@kaizen/component-library/icons/bulletted-list.icon.svg"
+import numberedListIcon from "@kaizen/component-library/icons/numbered-list.icon.svg"
+import decreaseIndentIcon from "@kaizen/component-library/icons/decrease-indent.icon.svg"
+import increaseIndentIcon from "@kaizen/component-library/icons/increase-indent.icon.svg"
+
+import { EditorState, Transaction } from "prosemirror-state"
+import { Schema, NodeType, MarkType } from "prosemirror-model"
+import { Command, toggleMark } from "prosemirror-commands"
+import { wrapInList, liftListItem, sinkListItem } from "prosemirror-schema-list"
+import { markIsActive } from "@cultureamp/rich-text-toolkit"
+import { ToolbarItems, ToolbarControlTypes } from "./RichTextEditor"
+
+type ToolbarControl = {
+  icon: React.SVGAttributes<SVGSymbolElement>
+  label: string
+  isActive: boolean
+  action: (
+    state: EditorState<any>,
+    dispatch: ((tr: Transaction<any>) => void) | undefined
+  ) => boolean
+}
+interface GroupedToolbarControls {
+  [group: string]: ToolbarControl[]
+}
+
+type ControlGroupTypes = {
+  [key in ToolbarControlTypes]?: string
+}
+
+function toggleMarkCommand(mark: MarkType): Command {
+  return (
+    state: EditorState,
+    dispatch: ((tr: Transaction) => void) | undefined
+  ) => toggleMark(mark)(state, dispatch)
+}
+
+function toggleListCommand(node: NodeType): Command {
+  return (
+    state: EditorState,
+    dispatch: ((tr: Transaction) => void) | undefined
+  ) => wrapInList(node)(state, dispatch)
+}
+
+function liftOrIndentList(action: "lift" | "indent"): Command {
```

Comment:
`liftOrIndentList` was written me a of helper function that leverages liftListItem and sinkListItem from prosemirror - the problem with liftListItem  or sinkListItem is they couldn't seem to determine what the current node was without it being passed in manually.

I'm re-evaluating combining them into one after start to wonder about the active and inactive state we want to add into these buttons so it might be better to break them apart anyway, even if it is a little more verbose